Mathlib/LinearAlgebra/AffineSpace/Basis.lean

```diff
@@ -252,6 +252,50 @@ noncomputable def coords : P →ᵃ[k] ι → k where
 theorem coords_apply (q : P) (i : ι) : b.coords q i = b.coord i q :=
   rfl
 
+/-- The affine equivalence between an affine space with a basis and the space of
+barycentric coordinates.
+
+**TODO**: The module docstring requests an equivalence `P ≃ᵃ[k] { f : ι →₀ k | f.sum = 1 }`.
+This is the correct mathematical statement: the set `{ f : ι → k | ∑ i, f i = 1 }` is an
+affine hyperplane in `ι → k` (an affine subspace with direction `{ f | ∑ i, f i = 0 }`).
+
+However, encoding this as a type in Lean is nontrivial. We would need to either:
+1. Define it as an `AffineSubspace` and show the subtype inherits affine space structure, or
+2. Directly construct the affine space structure on the subtype.
```

Comment:
The mathlib style isn't to use something that is *syntactically* `{ f : ι →₀ k | f.sum = 1 }`. Rather, you construct an `AffineSubspace` (as a named `def`) and then show that it has that set of points. And you can construct that affine subspace using `AffineSubspace.comap`; that is, it's the preimage of the singleton subspace of `k` (considered as an affine space over itself) that contains just the point `1` (which can be expressed as the affine span of that one point) under the linear map (converted to an affine map with `LinearMap.toAffineMap`) that sends a `Finsupp` to the sum of its values (and I think that linear map is `Finsupp.linearCombination k (1 : ι → k)`, though there may be other ways to express it as well).